### DISCUSSION TITLE
API requests are available via audit log streaming – Public Beta

### DISCUSSION THREAD

**Discussion by tuves** [community/community#51940] [Enterprise]
GitHub is the software development platform trusted by millions of developers. With this trust comes a responsibility to constantly improve our security, audit, and compliance solutions with the developer in mind. A high-quality [audit log](https://docs.github.com/enterprise-cloud@latest/admin/monitoring-activity-in-your-enterprise/reviewing-audit-logs-for-your-enterprise/about-the-audit-log-for-your-enterprise) is an essential tool for enterprises to ensure compliance, maintain security, investigate issues, and promote accountability. GitHub has been working hard to add new features to help enterprises gain more visibility into their organizations’ software development and collaboration activities, and operate with greater peace of mind when it comes to security and compliance.

GitHub Enterprise Cloud customers can now join a public beta for streaming API request events as part of their enterprise audit log.

As part of this beta, REST API calls against enterprise's private and internal repositories can be streamed to one of [GitHub's supported streaming endpoints](https://docs.github.com/en/enterprise-cloud@latest/admin/monitoring-activity-in-your-enterprise/reviewing-audit-logs-for-your-enterprise/streaming-the-audit-log-for-your-enterprise).

Check out our [changelog](https://github.blog/changelog/2023-04-03-api-requests-are-available-via-audit-log-streaming-public-beta/) for instructions to enable the feature and [blog](https://github.blog/2023-04-03-level-up-monitoring-and-reporting-for-your-enterprise/) for more information!

**Comment by Kazy0004** [comment]
مرحبا اود مسعدتك في معرفه الكثير التي تخصني في ادارة شؤون github

**Comment by ManikOfficial** [comment]
Nice

**Reply by mazr711001** [reply]
Please release

**Reply by WOAftp** [reply]
Hugh?

WOAftp 

Our AMA  or IAMARAMP

**Reply by bluesky-dev2** [reply]
Ok. I think so

**Comment by tonglil** [comment]
Can you consider providing a per-repo audit log interface so that repo owners (like teams) can see what happened in a repo, such as branch protection rules being changed, without needing org owner access?

**Reply by boylejj** [reply]
👋🏻  @tonglil, audit log PM here. This is something that we can consider. One alternative we have been considering is having the ability to assign a read audit log role to non-org owners? This would allow users to be assigned a "read audit log" role by the organization owner, allowing them to query the organization's audit logs without org owner permissions. Would this be a slightly different way of addressing the feedback you've provided? Please let me know what you think.

**Reply by tonglil** [reply]
That's a fine solution, however it would be better to grant audit logs to individual repos based on their roles in their team.

For example, the admins (or other role) of a repo can view the audit logs of that repo.

Viewing it on the repo-level also provides granularity that's scoped to that repo (essentially prefixes the repo query) so it's much more digestible. This is a perspective coming from an org with thousands of repos, and right now typing repos in is really slow in the UI.

FWIW, I know GH is trying to get search interfaces to look and act the same, but the audit log UI/UX is not that great, particularly when you want to see a timeline of what happened, or figure out if X changed. The query names are sometimes non-intuitive and there's no mapping (that I've been able to find in docs) that correlates what to what in the repo's setting page.

**Reply by boylejj** [reply]
Would you be willing to meet with me to discuss your Org/Repository use case, and better understand how the Audit Log U/I could be refactored to better support the types of searches you're typically performing?

**Reply by tonglil** [reply]
I'm happy to share my feedback.

**Comment by Anonymes1** [comment]
> GitHub is the software development platform trusted by millions of developers. With this trust comes a responsibility to constantly improve our security, audit, and compliance solutions with the developer in mind. A high-quality [audit log](https://docs.github.com/enterprise-cloud@latest/admin/monitoring-activity-in-your-enterprise/reviewing-audit-logs-for-your-enterprise/about-the-audit-log-for-your-enterprise) is an essential tool for enterprises to ensure compliance, maintain security, investigate issues, and promote accountability. GitHub has been working hard to add new features to help enterprises gain more visibility into their organizations’ software development and collaboration activities, and operate with greater peace of mind when it comes to security and compliance.
> 
> GitHub Enterprise Cloud customers can now join a public beta for streaming API request events as part of their enterprise audit log.
> 
> As part of this beta, REST API calls against enterprise's private and internal repositories can be streamed to one of [GitHub's supported streaming endpoints](https://docs.github.com/en/enterprise-cloud@latest/admin/monitoring-activity-in-your-enterprise/reviewing-audit-logs-for-your-enterprise/streaming-the-audit-log-for-your-enterprise).
> 
> Check out our [changelog](https://github.blog/changelog/2023-04-03-api-requests-are-available-via-audit-log-streaming-public-beta/) for instructions to enable the feature and [blog](https://github.blog/2023-04-03-level-up-monitoring-and-reporting-for-your-enterprise/) for more information!

> 
```

```

**Comment by Anonymes1** [comment]
Hackers

**Reply by Anonymes1** [reply]
@bluesky-dev2

**Comment by kumavis** [comment]
Feature Request:
![image](https://user-images.githubusercontent.com/1474978/235001034-71589f50-c966-43f5-8dbe-889f66d4b42f.png)
a button here (or similar) for those with access to the audit log to navigate to the audit logs filtered for this repo

**Comment by rumon5h** [comment]
Just awesome

**Comment by gdcraigmunday** [comment]
Does this new feature also include the streaming of audit events for package reads?  That is, when a user downloads a package from a repository?
